torch/_prims_common/__init__.py

```diff
@@ -77,11 +77,16 @@
 TensorOrNumberLikeType = Union[TensorLikeType, NumberType]
 
 
-def same_shape(a: ShapeType, b: ShapeType) -> bool:
+def same_shape(a: ShapeType, b: ShapeType, *, allow_rhs_unbacked=False) -> bool:
     if len(a) != len(b):
         return False
 
     for x, y in zip(a, b):
+        if allow_rhs_unbacked:
+            # TODO: We should check that the symbols are consistent
+            # with each other
+            if isinstance(y, torch.SymInt):
+                continue
```

Comment:
What's the idea here?

Reply:
If I run a data-dependent op, I will end up with an unbacked symint on the fake tensor. I cannot directly compare the unbacked symint with the real symint. So I just say "well, whatever, if the fake value is symint, let's just assume it's fine." It's definitely possible to do a more strict check but I was lazy.